public/youtube-request.js

```diff
@@ -0,0 +1,41 @@
+/* global gapi */
+
+/**
+ * Sample JavaScript code for youtube.search.list
+ * See instructions for running APIs Explorer code samples locally:
+ * https://developers.google.com/explorer-help/guides/code_samples#javascript
+ */
+
+function loadClient(key) {
+  gapi.client.setApiKey(key);
+  return gapi.client
+    .load('https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest')
+    .then(
+      function () {
+        console.log('GAPI client loaded for API');
+      },
+      function (err) {
+        console.error('Error loading GAPI client for API', err);
+      }
+    );
+}
+// Make sure the client is loaded before calling this method.
+function execute(search) {
+  return gapi.client.youtube.search
```

Comment:
What if gapi is not loaded yet by the time your components needs to use it? You might want to use youtube API inside a hook in react.